File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> numbers, string direction) {
    if(direction[0] == 'r') {
        rotate(numbers.rbegin(), numbers.rbegin() + 1, numbers.rend());
    } else if(direction[0] == 'l') {
        rotate(numbers.begin(), numbers.begin() + 1, numbers.end());
    }
    return numbers;
}

File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.swift
import Foundation

func solution(_ numbers:[Int], _ direction:String) -> [Int] {
    // S1
    var result = numbers
    if direction == "right" {
        let lastElement = result.removeLast()
        result.insert(lastElement, at: 0)  // [lastElement] + result
    } else {
        let firstElement = result.removeFirst()
        result.append(firstElement)
    }
    
    return result
    
    // S2
    // if direction == "right" {
    //     return [numbers.last!] + numbers.dropLast()
    // } else {
    //     return Array(numbers.dropFirst()) + [numbers.first!]
    // }
    
    
}